app/models/carto_uploader.rb

```diff
@@ -33,29 +52,92 @@ def upload(data)
       end
       keys = keys.join(",")
       vals = vals.join(",")
-      allVals.push("(#{vals})");  #For Mass Insertion mode only.
+      arrVals.push("(#{vals})")
       #--------------------------------
-
-      #Option 1: Individual SQL Insertions
+      
+      #If we have enough classifications queued up, do a batch Insert.
       #--------------------------------
-      if @mode != CARTODB_INSERT_MODE_MASS
-        sqlQuery = "INSERT INTO #{CARTODB_TABLE} (#{keys}) VALUES (#{vals}) "
-        uri = URI(CARTODB_URI_GET.gsub(/__SQLQUERY__/, URI.escape(sqlQuery)))
-        res = Net::HTTP.get(uri)
-        puts res
+      if (arrVals.length >= CLASSIFICATIONS_PER_BATCH)
+        expectedInsertions += arrVals.length
+        successfulInsertions += batchInsert(keys, arrVals)
+        arrVals = []
+        #Sanity check
+        puts "Inserted #{successfulInsertions} Classifications out of #{expectedInsertions}"
+        if (expectedInsertions != successfulInsertions)
+          #TODO
+        end
       end
       #--------------------------------
     end
 
-    #Option 2: Mass SQL Insertions
+    #Do a final batch Insert for any stragglers
+    #--------------------------------
+    if (arrVals.length > 0)
+      expectedInsertions += arrVals.length
+      successfulInsertions += batchInsert(keys, arrVals)
+      arrVals = []
+    end
+    #--------------------------------
+    
+    #Final Sanity check
     #--------------------------------
-    allVals = allVals.join(",")
-    if @mode == CARTODB_INSERT_MODE_MASS
+    puts "Inserted #{successfulInsertions} Classifications out of #{expectedInsertions}"
+    if (expectedInsertions != successfulInsertions)
+      #TODO
+    end
+    #--------------------------------
+  end
+  
+  #Returns latest Classification ID.
+  def getLatestClassificationID()
+    begin
+      sqlQuery = "SELECT classification_id FROM #{CARTODB_TABLE} ORDER BY classification_id DESC LIMIT 1"
+      uri = URI(CARTODB_URI_GET.gsub(/__SQLQUERY__/, URI.escape(sqlQuery)))
```

Comment:
I've taken care of `snake_case`s, superfluous `initialize`s and unnecessary `begin`s, but I'd like to get more thoughts on the duplication of the API calls. Right now there are only 3 methods where the API is called, and the only commonality is with the two `GET` requests.

```
uri = URI(CARTODB_URI_GET.gsub(/__SQLQUERY__/, URI.escape(sql_query)))
res = Net::HTTP.get(uri)
res = JSON.parse(res)
if (res["error"])
  raise StandardError, res["error"]
end
```

I'm not sure if this is worth extracting into a separate function, but I may be reading your comment in the wrong way - thoughts?